packages/framework/src/Form/Admin/Advert/AdvertFormType.php

```diff
@@ -39,16 +40,24 @@ class AdvertFormType extends AbstractType
      */
     private $advertPositionRegistry;
 
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Image\Config\ImageConfig
+     */
+    private $imageConfig;
+
     /**
      * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
      * @param \Shopsys\FrameworkBundle\Model\Advert\AdvertPositionRegistry $advertPositionRegistry
+     * @param \Shopsys\FrameworkBundle\Component\Image\Config\ImageConfig $imageConfig
      */
     public function __construct(
         Domain $domain,
-        AdvertPositionRegistry $advertPositionRegistry
+        AdvertPositionRegistry $advertPositionRegistry,
+        ImageConfig $imageConfig
```

Comment:
Adding a new parameter to a constructor is a BC break according to our BC promise.
You can make this nullable with default null value and add setter injection.
```php
    ?ImageConfig $imageConfig = null 
```

``` php
/**
     * @required
     * @internal This function will be replaced by constructor injection in next major
     * @param \Shopsys\FrameworkBundle\Component\Cron\CronModuleExecutor $cronModuleExecutor
     */
    public function setCronModuleExecutor(CronModuleExecutor $cronModuleExecutor)
    {
        if ($this->cronModuleExecutor !== null && $this->cronModuleExecutor !== $cronModuleExecutor) {
            throw new BadMethodCallException(sprintf('Method "%s" has been already called and cannot be called multiple times.', __METHOD__));
        }
        if ($this->cronModuleExecutor === null) {
            @trigger_error(sprintf('The %s() method is deprecated and will be removed in the next major. Use the constructor injection instead.', __METHOD__), E_USER_DEPRECATED);
            $this->cronModuleExecutor = $cronModuleExecutor;
        }
    }
```

Reply:
Thx. DI has been change to setter